crates/kas-core/src/event/cx/key.rs

```diff
@@ -447,9 +457,37 @@ impl<'a> EventCx<'a> {
                 self.send_event(widget.re(), id.clone(), Event::KeyFocus);
             }
 
-            if ime.is_some() && (ime != self.ime || target_is_new) {
-                window.set_ime_allowed(ime);
-                self.ime = ime;
+            if let Some(purpose) = ime
+                && self.ime.is_none()
+            {
+                let capabilities = ImeCapabilities::new()
+                    .with_hint_and_purpose()
+                    .with_cursor_area();
+
+                let hint = ImeHint::empty(); // TODO
+
+                // NOTE: we provide bogus cursor area and update in `frame_update`;
+                // the API does not allow to only provide this later.
+                let position = LogicalPosition::new(0, 0);
+                let size = LogicalSize::new(0, 0);
```

Comment:
We are now required to set size & position immediately even though we will update this before the first frame, so the values provided here shouldn't matter.